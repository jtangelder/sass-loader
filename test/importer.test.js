import sass from 'sass';
import resolve from 'enhanced-resolve';

import createSassImporter from '../src/importer';

describe('importer', () => {
  it('should resolve imports when passed to `sass`', (done) => {
    const importer = createSassImporter(resolve.create);

    sass.render(
      {
        file: 'test/sass/imports.sass',
        importer,
      },
      (err, result) => {
        expect(result.css.toString()).toMatchSnapshot();

        done(err);
      }
    );
  });
});
