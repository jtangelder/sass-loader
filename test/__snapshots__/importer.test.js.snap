// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importer should resolve imports when passed to \`sass\` 1`] = `
"@charset \\"UTF-8\\";
/* @import another/module */
@import url(http://example.com/something/from/the/interwebs);
.another-sass-module {
  background: hotpink;
}

/* @import another/underscore */
.underscore {
  background: hotpink;
}

/* @import another/_underscore */
.underscore {
  background: hotpink;
}

/* @import ~sass/underscore */
.underscore-sass {
  background: hotpink;
}

/* @import ~sass/some.module */
.some-sass-module {
  background: hotpink;
}

/* @import url(http://example.com/something/from/the/interwebs); */
/* scoped import @import language */
.scoped-import body {
  font: 100% Helvetica, sans-serif;
  color: #333;
}
.scoped-import nav ul {
  margin: 0;
  padding: 0;
  list-style: none;
}
.scoped-import nav li {
  display: inline-block;
}
.scoped-import nav a {
  display: block;
  padding: 6px 12px;
  text-decoration: none;
}
.scoped-import .box {
  -webkit-border-radius: 10px;
  -moz-border-radius: 10px;
  -ms-border-radius: 10px;
  border-radius: 10px;
}
.scoped-import .message, .scoped-import .warning, .scoped-import .error, .scoped-import .success {
  border: 1px solid #ccc;
  padding: 10px;
  color: #333;
}
.scoped-import .success {
  border-color: green;
}
.scoped-import .error {
  border-color: red;
}
.scoped-import .warning {
  border-color: yellow;
}
.scoped-import .foo:before {
  content: \\"\\";
}
.scoped-import .bar:before {
  content: \\"∑\\";
}

/* @import util */
.util {
  color: hotpink;
}

/* @import ~module */
.module {
  background: hotpink;
}

/* @import ~another */
.another-scss-module-from-node-modules {
  background: hotpink;
}

a {
  color: red;
}"
`;
